src/bench/bench.cpp

```diff
@@ -136,31 +146,62 @@ void BenchRunner::RunAll(const Args& args)
             continue;
         }
 
-        Bench bench;
-        if (args.sanity_check) {
-            bench.epochs(1).epochIterations(1);
-            bench.output(nullptr);
-        }
-        bench.name(name);
-        g_running_benchmark_name = name;
-        if (args.min_time > 0ms) {
-            // convert to nanos before dividing to reduce rounding errors
-            std::chrono::nanoseconds min_time_ns = args.min_time;
-            bench.minEpochTime(min_time_ns / bench.epochs());
-        }
+        for (int threads : thread_scales) {
+            std::vector<std::string> current_setup_args = args.setup_args;
+            if (threads > 0) {
+                bool found = false;
+                for (auto& arg : current_setup_args) {
+                    if (arg.starts_with("-worker-threads=") == 0) {
```

Comment:
hmmm, this looks like a refactor gone wrong, was this `arg.find("-worker-threads=") == 0` originally?

```suggestion
                    if (arg.starts_with("-worker-threads=")) {
```